From 85d3e1e13fe2728f4b0d940d609867cccd48a847 Mon Sep 17 00:00:00 2001
From: Prithvi A Pai <prithvi.a.pai@intel.com>
Date: Wed, 6 Dec 2023 17:01:25 +0530
Subject: [PATCH] Certificate buffer overflow fix

libspdm had the certificate buffer limitation of 4k.
When the certificate data increases the Certificate verification fails.
Hence added patch to define certificate length at build time.

Tested:
BMC as requester is able to verify the certificate of responders

Signed-off-by: Prithvi A Pai <prithvi.a.pai@intel.com>
---
 include/library/spdm_lib_config.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/library/spdm_lib_config.h b/include/library/spdm_lib_config.h
index 6203b51f13..26c478c6ed 100644
--- a/include/library/spdm_lib_config.h
+++ b/include/library/spdm_lib_config.h
@@ -63,7 +63,7 @@
  * libspdm context.
  */
 #ifndef LIBSPDM_MAX_CERT_CHAIN_SIZE
-#define LIBSPDM_MAX_CERT_CHAIN_SIZE 0x1000
+#define LIBSPDM_MAX_CERT_CHAIN_SIZE 0x1400
 #endif
 #ifndef LIBSPDM_MAX_MEASUREMENT_RECORD_SIZE
 #define LIBSPDM_MAX_MEASUREMENT_RECORD_SIZE 0x1000
@@ -73,7 +73,7 @@
  * in bytes, of a partial certificate that can be sent or received.
  */
 #ifndef LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN
-#define LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN 1024
+#define LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN 1000
 #endif
 
 /* To ensure integrity in communication between the Requester and the Responder libspdm calculates
-- 
2.34.1

